### PR TITLE
Add the possibility for asv machine to use defaults

### PR DESCRIPTION
**Pull request by Lothiraldan**
Using asv in a full-automated way is difficult because asv run needs asv
machine to run first. Make asv machine accept a '--yes' that accept the
default value. This way automated scripts can put 'asv machine --yes' before
running 'asv run'.

### CODE REVIEW

File: asv/console.py [hunk 1]
```diff
@@ -230,8 +230,12 @@ def color_print(*args, **kwargs):
         write(end)
 
 
-def get_answer_default(prompt, default):
+def get_answer_default(prompt, default, accept_default=False):
     color_print("{0} [{1}]: ".format(prompt, default), end='')
+
+    if accept_default is True:
```
Comment (pv) [inline asv/console.py]: `if accept_default` according to pep8

File: asv/machine.py [hunk 2]
```diff
@@ -147,8 +147,8 @@ def get_defaults():
             }
 
     @staticmethod
-    def generate_machine_file():
-        if not sys.stdout.isatty():
+    def generate_machine_file(accept_default=False):
+        if not sys.stdout.isatty() and accept_default is False:
```
Comment (pv) [inline asv/machine.py]: `not accept_default`

**Comment by pv** [comment]
lgtm, aside from style nitpicks

**Comment by pv** [comment]
Also, adding a test in `test/test_machine.py` for this would be useful.
You can use `run_asv` from `test/tools.py` to run asv in a test.

**Comment by Lothiraldan** [comment]
@pv Sorry for the delay, I added a unit-test at the Machine object level. Do you think it's sufficient ?

**Comment by pv** [comment]
thanks, merged

**Pull request merged**